ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/monitor-sessions.j2

```diff
@@ -0,0 +1,57 @@
+{% if monitor_sessions is arista.avd.defined %}
+
+## Monitor Sessions
+
+### Monitor Sessions Summary
+{%     for monitor_session in monitor_sessions | arista.avd.natural_sort('name') if monitor_session.name is arista.avd.defined %}
+{%         if monitor_session.sources is arista.avd.defined and monitor_session.destinations is arista.avd.defined %}
+
+#### {{ monitor_session.name }}
+
+##### {{ monitor_session.name }} Sources
+
+| Sources | Direction | Access Group Type | Access Group Name | Access Group Priority |
+| ------- | --------- | ----------------- | ----------------- | --------------------- |
+{%             for source in monitor_session.sources | arista.avd.natural_sort('name') if source.name is arista.avd.defined %}
+| {{ source.name }} | {{ source.direction | arista.avd.default('both') }} | {{ source.access_group.type | arista.avd.default('-') }} | {{ source.access_group.name | arista.avd.default('-') }} | {{ source.access_group.priority | arista.avd.default('-') }} |
+{%             endfor %}
+
+##### {{ monitor_session.name }} Destinations and Session Settings
+
+| Settings | Values |
+| -------- | ------ |
+| Destinations | {{ monitor_session.destinations | join(', ') }} |
+{%             if monitor_session.encapsulation_gre_metadata_tx is arista.avd.defined(true) %}
+| Encapsulation Gre Metadata Tx | {{ monitor_session.encapsulation_gre_metadata_tx }} |
+{%             endif %}
+{%             if monitor_session.header_remove_size is arista.avd.defined %}
+| Header Remove Size | {{ monitor_session.header_remove_size }} |
+{%             endif %}
+{%             if monitor_session.access_group.type is arista.avd.defined and monitor_session.access_group.name is arista.avd.defined %}
+| Access Group Type | {{ monitor_session.access_group.type }} |
+| Access Group Name | {{ monitor_session.access_group.name }} |
+{%             endif %}
+{%             if monitor_session.rate_limit_per_ingress_chip is arista.avd.defined %}
+| Rate Limit per Ingress Chip | {{ monitor_session.rate_limit_per_ingress_chip }} |
+{%             endif %}
+{%             if monitor_session.rate_limit_per_ingress_chip is arista.avd.defined %}
+| Rate Limit per Egress Chip | {{ monitor_session.rate_limit_per_egress_chip }} |
+{%             endif %}
+{%             if monitor_session.sample is arista.avd.defined %}
+| Sample | {{ monitor_session.sample }} |
+{%             endif %}
+{%             if monitor_session.truncate.enabled is arista.avd.defined(true) and monitor_session.truncate.size is arista.avd.defined %}
```

Comment:
Looks good. I have a suggestion to replace lines 43-48:

```
 {%             if monitor_session.truncate.enabled is arista.avd.defined(true) %}
| Truncate Enabled | {{ monitor_session.truncate.enabled }} |
{%                 if monitor_session.truncate.size is arista.avd.defined %}
| Truncate Size | {{ monitor_session.truncate.size }} |
{%                 endif %}
{%             endif %}
```

Reply:
thanks!! i'm still thinking too complicated sometimes